app/src/ui/merge-branch/merge.tsx

```diff
@@ -143,7 +149,7 @@ export class Merge extends React.Component<IMergeProps, IMergeState> {
               Merge into <strong>{currentBranch ? currentBranch.name : ''}</strong>
             </Button>
           </ButtonGroup>
-          {mergeInfo}
+          {this.renderMergeInfo(invalidBranch)}
```

Comment:
Yes, I am suggesting `{this.renderMergeInfo()}` be called and the check for invalid branch be placed there. Besides following established patterns, it keeps the logic for showing a specific footer within the method itself.